include/aws/common/common.h

```diff
@@ -117,19 +119,26 @@
  * Violations of the function contracts are undefined behaviour.
  */
 #ifdef CBMC
-#define AWS_PRECONDITION(cond) __CPROVER_assume(cond)
-#define AWS_POSTCONDITION(cond) AWS_ASSERT(cond)
+#define AWS_PRECONDITION_2(cond, explanation) __CPROVER_precondition((cond), (explanation))
+#define AWS_PRECONDITION_1(cond, explanation) __CPROVER_precondition((cond), "")
+#define AWS_POSTCONDITION_2(cond, explanation) __CPROVER_assert((cond), (explanation))
+#define AWS_POSTCONDITION_1(cond, explanation) __CPROVER_assert((cond), "")
```

Comment:
`__CPROVER_assert((cond), #cond " check failed")` ?

Also, I think you should be able to remove explanation from the _1 variants.

That way you get an explanation, it's just exactly the code that failed